pkg/driver/deployment/deployer.go

```diff
@@ -0,0 +1,40 @@
+package deployment
+
+import (
+	"github.com/sirupsen/logrus"
+	"github.com/vhive-serverless/loader/pkg/common"
+	"github.com/vhive-serverless/loader/pkg/config"
+)
+
+type FunctionDeployer interface {
+	Deploy(functions []*common.Function, configuration interface{})
+	Clean()
+}
+
+func CreateDeployer(cfg *config.Configuration) (deployer FunctionDeployer, configuration interface{}) {
+	switch cfg.LoaderConfiguration.Platform {
+	case "Knative", "Knative-RPS":
+		deployer = &knativeDeployer{}
+		configuration = knativeDeploymentConfiguration{
+			YamlPath:          cfg.YAMLPath,
+			IsPartiallyPanic:  cfg.LoaderConfiguration.IsPartiallyPanic,
+			EndpointPort:      cfg.LoaderConfiguration.EndpointPort,
+			AutoscalingMetric: cfg.LoaderConfiguration.AutoscalingMetric,
+		}
+	case "OpenWhisk", "OpenWhisk-RPS":
+		deployer = &openWhiskDeployer{}
+		configuration = openWhiskDeploymentConfiguration{}
+	case "AWSLambda", "AWSLambda-RPS":
```

Comment:
```go
return NewAWSLambdaDeployement()
```

You can then return like this and do this the same for every deployer. Then this fine can even be outside of the package.